streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -713,6 +735,97 @@ boolean runLoop() {
         return true;
     }
 
+    // visible for testing
+    void maybeGetClientInstanceIds() {
+        // we pass in a timeout of zero into each `clientInstanceId()` call
+        // to just trigger the "get instance id" background RPC;
+        // we don't want to block the stream thread that can do useful work in the meantime
+
+        if (fetchDeadline != -1) {
+            if (!mainConsumerInstanceIdFuture.isDone()) {
+                if (fetchDeadline >= time.milliseconds()) {
+                    try {
+                        mainConsumerClientInstanceId = mainConsumer.clientInstanceId(Duration.ZERO);
+                        mainConsumerInstanceIdFuture.complete(mainConsumerClientInstanceId);
+                        maybeResetFetchDeadline();
+                    } catch (final IllegalStateException disabledError) {
+                        mainConsumerInstanceIdFuture.complete(null);
```

Comment:
We might want to add logging, similar to admin case... (same for others below) -- forgot to do this.

Was also wondering if KS code should log client instance id on successful retrieval? Ask a similar question on another KIP-714 PR for clients.